### PR TITLE
feat: add C implementation for `math/base/assert/is-nonnegative-integer`

### LINKED ISSUE #1723: [RFC]: Add C implementation for `@stdlib/math/base/assert/is-nonnegative-integer`
### Description

This RFC proposes adding C implementation for `@stdlib/math/base/assert/is-nonnegative-integer`.

### Related Issues

Related issues #649.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves #1723 

## Description
Added C implementation for [`stdlib/math/base/assert/is-nonnegative-integer`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-nonnegative-integer).

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for `stdlib/math/base/assert/is-nonnegative-integer`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes #1723 
-   pre-requisite for #1664 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/addon.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_is_nonnegative_integer )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/addon.c]: This isn't correct, as `D_D` corresponds to double-in and double-out. We should probably add `D_X` to `math/base/napi/unary` as a separate PR and then update this PR accordingly.

File: lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+*/
+
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
+#include "stdlib/math/base/special/floor.h"
+
+/**
+* Tests if a finite double-precision floating-point number is a nonnegative integer.
+*
+* @param x    input value
+* @return	  output value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c]: This line has mixed tabs and spaces.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks, @gunjjoshi. Left a couple of comments. Looks like we need to update `math/base/napi/unary` OR you can manually handle the add-on interface. See, e.g., https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/assert/is-even/src/addon.c.

File: lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/addon.c (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/addon.c]: Left a follow-up comment below. The other alternative is to manually perform argument munging.

**Comment by gunjjoshi** [comment]
> Thanks, @gunjjoshi. Left a couple of comments. Looks like we need to update `math/base/napi/unary` OR you can manually handle the add-on interface. See, e.g., https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/assert/is-even/src/addon.c.

For now, I've manually handled the add-on interface. 
However, I will try to make a separate PR adding `D_X` to `math/base/napi/unary`.
Thanks for pointing this out.

**Comment by gunjjoshi** [comment]
Hey @kgryte @Planeshifter, any other suggestions on this ?

File: lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+*/
+
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
+#include "stdlib/math/base/special/floor.h"
+
+/**
+* Tests if a finite double-precision floating-point number is a nonnegative integer.
+*
+* @param x	input value
+* @return	output value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

You should use spaces to align descriptions in C files.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @gunjjoshi!

File: lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c (hunk 3 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/main.c]: Thanks for pointing this out. I will keep that in mind.

**Pull request merged**